Code/Editor/Controls/FolderTreeCtrl.cpp

```diff
@@ -16,87 +15,76 @@
 #include <QSortFilterProxyModel>
 #include <QStandardItemModel>
 
+#include <AzCore/std/algorithm.h>
 // AzQtComponents
-#include <AzQtComponents/Utilities/DesktopUtilities.h>      // for AzQtComponents::ShowFileOnDesktop
-
-
-enum ETreeImage
-{
-    eTreeImage_Folder = 0,
-    eTreeImage_File = 2
-};
-
-enum CustomRoles
-{
-    IsFolderRole = Qt::UserRole
-};
+#include <AzQtComponents/Utilities/DesktopUtilities.h> // for AzQtComponents::ShowFileOnDesktop
 
 //////////////////////////////////////////////////////////////////////////
 // CFolderTreeCtrl
 //////////////////////////////////////////////////////////////////////////
-CFolderTreeCtrl::CFolderTreeCtrl(const QStringList& folders, const QString& fileNameSpec,
-    const QString& rootName, bool bDisableMonitor, bool bFlatTree, QWidget* parent)
-    : QTreeView(parent)
-    , m_rootTreeItem(nullptr)
-    , m_folders(folders)
-    , m_fileNameSpec(fileNameSpec)
-    , m_rootName(rootName)
-    , m_bDisableMonitor(bDisableMonitor)
-    , m_bFlatStyle(bFlatTree)
-{
-    init(folders, fileNameSpec, rootName, bDisableMonitor, bFlatTree);
-}
 
 CFolderTreeCtrl::CFolderTreeCtrl(QWidget* parent)
     : QTreeView(parent)
-    , m_rootTreeItem(nullptr)
-    , m_bDisableMonitor(false)
-    , m_bFlatStyle(true)
-
+    , m_folderIcon(":/TreeView/folder-icon.svg")
+    , m_fileIcon(":/TreeView/default-icon.svg")
+    , m_model(new QStandardItemModel(this))
+    , m_proxyModel(new QSortFilterProxyModel(this))
 {
-}
+    m_proxyModel->setSourceModel(m_model);
+    m_proxyModel->setRecursiveFilteringEnabled(true);
+    setModel(m_proxyModel);
 
+    QObject::connect(this, &QTreeView::doubleClicked, this, &CFolderTreeCtrl::OnIndexDoubleClicked);
+}
 
-void CFolderTreeCtrl::init(const QStringList& folders, const QString& fileNameSpec, const QString& rootName, bool bDisableMonitor /*= false*/, bool bFlatTree /*= true*/)
+void CFolderTreeCtrl::Configure(
+    const AZStd::vector<QString>& folders, const QString& fileNameSpec, const QString& rootName, bool bDisableMonitor, bool bFlatTree)
 {
-    m_model = new QStandardItemModel(this);
-    m_proxyModel = new QSortFilterProxyModel(this);
-    m_proxyModel->setRecursiveFilteringEnabled(true);
-
-    m_proxyModel->setSourceModel(m_model);
-    setModel(m_proxyModel);
+    m_folders.clear();
+    AZStd::ranges::copy_if(
+        folders,
+        AZStd::back_inserter(m_folders),
+        [](const auto& path)
+        {
+            if (CFileUtil::PathExists(path) || Path::IsFolder(path.toLocal8Bit().constData()))
+            {
+                return true;
+            }
```

Comment:
Suggestion: Consider using AZ::IO::SystemFile::Exists(path) and IsDirectory(path) instead, it would be nice if we could eventually deprecate FileUtil / PathUtil since they overlap with AzCore/AzFramework methods.